codec/reflectcodec/type_codec.go

```diff
@@ -279,13 +300,18 @@ func (c *genericCodec) MarshalInto(value interface{}, p *wrappers.Packer) error
 		return errMarshalNil // can't marshal nil
 	}
 
-	return c.marshal(reflect.ValueOf(value), p, c.maxSliceLen)
+	metaValue := reflect.ValueOf(value)
+	if metaValue.Kind() == reflect.Ptr {
+		metaValue = metaValue.Elem()
+	}
```

Comment:
What is this change doing? And is it required?

Reply:
Replied above ^